BootloaderCommonPkg/Include/Library/ContainerLib.h

```diff
@@ -33,7 +35,16 @@ typedef UINT8 AUTH_TYPE;
 // Attributes for COMPONENT_ENTRY
 #define COMPONENT_ENTRY_ATTR_RESERVED       BIT7
 
-typedef VOID (*LOAD_COMPONENT_CALLBACK) (UINT32 ProgressId);
+
+typedef struct {
+  UINT32           ComponentType;
+  UINT8           *CompBuf;
+  UINT32           CompLen;
+  HASH_ALG_TYPE    HashAlg;
+  UINT8           *HashData;
+} COMPONENT_CB_INFO;

```

Comment:
Better use COMPONENT_CALLBACK_INFO if you have other things to update this patch.